wiggum_app/controller.py

```diff
@@ -286,10 +303,15 @@ def main():
                 rank_param = agg_type + '_view_' + view_score
                 rank_result = labeled_df_setup.rank_occurences_by_view(rank_param,view_score)
 
-            result_dict_dict = {}
-            result_dict_dict = models.getResultDict(labeled_df_setup, rank_result)
+            # Generate distance heatmaps
+            distance_heatmap_dict = models.getDistanceHeatmapDict(rank_result)
+
+            df = labeled_df_setup.df.to_dict(orient='records')
+            df = json.dumps(df, indent=2)
 
-            return jsonify(result_dict_dict)
+            return jsonify(distance_heatmap_dict = distance_heatmap_dict, 
```

Comment:
if filters have been applied they should still be applied after ranking. so maybe re-filter the list? after ranking if the table is filtered when the button is pushed?

Reply:
If filter_result can be stored in LabeledDataFrame, I can use it for ranking. If no filter, filter_result is equal to result_df. Or I can do filter process again before ranking in controller.py. Which one is good for now?

Reply:
I just tested on iris test dataset, click detect and then click rank button, didn't see any error. Can you give more detail about this error?

Reply:
If click "command" key and click the mouse at the same time in Mac, it can unselect the previous selected option.